weatherman.py

```diff
@@ -30,6 +33,7 @@ def map_months(years_month):
     for ym in years_month:
         if '/' in ym:
             year, month = ym.split('/')
+            month = month.lstrip('0')
```

Comment:
You are removing the leading zeros from month and then mapping it. It's the very reason that your code is working for e.g. `02` and `2` at the same time. How do you don't know about this while you wrote it by yourself?